push/src/gateway.cc

```diff
@@ -15,6 +16,31 @@ namespace prometheus {
 static const char CONTENT_TYPE[] =
     "Content-Type: text/plain; version=0.0.4; charset=utf-8";
 
+class CurlWrapper {
+public:
+  CurlWrapper() {
+    curl_ = nullptr;
+  }
+  ~CurlWrapper() {
+    std::lock_guard<std::mutex> l(mutex_);
+    if (curl_) {
+      curl_easy_cleanup(curl_);
+    }
+  }
+
+  CURL *curl() {
+    std::lock_guard<std::mutex> l(mutex_);
+    if (!curl_) {
+      curl_ = curl_easy_init();
```

Comment:
I'd move that into the ctor and throw if it fails

Reply:
As per contribution guide, code should follow Google's coding style guide where [it recommends avoiding initializations in constructors if they can fail, and forbids the use of exceptions](https://google.github.io/styleguide/cppguide.html#Doing_Work_in_Constructors).